src/service/utils/blockUtils.js

```diff
@@ -38,7 +38,11 @@ const uiBlockUser = async (postId, userId, source, reason, message, callback) =>
   try {
     await blockUser(data);
     Toast.show(handleMessage(reason, message), Toast.LONG);
-    if (callback) callback();
+    if (callback) {
+      setTimeout(() => {
+        callback();
+      }, 1000);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of `setTimeout` here introduces an artificial delay before invoking the callback. While this might be intended to improve user experience, it could potentially lead to confusion if the operation has actually completed but the UI doesn't reflect that immediately. If there's a need to wait for some operation to complete before calling the callback, it would be better to use promises or async/await to handle that.

If you still want to keep the delay, consider moving the delay logic to the caller of `uiBlockUser`, as it's more related to UI behavior rather than the blocking functionality itself.

```diff
-    if (callback) {
-      setTimeout(() => {
-        callback();
-      }, 1000);
-    }
+    if (callback) callback();
```
Then in the caller function:

```javascript
uiBlockUser(data, () => {
  setTimeout(() => {
    // Callback logic here
  }, 1000);
});
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->